src/signin.js

```diff
@@ -1287,3 +1289,26 @@ function newAppInstance (dom, appDetails, callback) {
   div.appendChild(b)
   return div
 }
+
+async function getUserRoles () {
+  const profile = await checkUser()
+  if (!profile) {
+    return []
+  }
+  const preferencesFile = UI.store.any(profile, ns.space('preferencesFile'), null, profile.doc())
+  if (!preferencesFile) {
+    return []
+  }
+  await UI.store.fetcher.load(preferencesFile)
+  return UI.store.each(profile, ns.rdf('type'), null, preferencesFile.doc())
+}
+
+async function filterAvailablePanes (panes) {
+  const userRoles = await getUserRoles()
+  return Object.values(panes).filter(pane => isMatchingAudience(pane, userRoles))
+
+  function isMatchingAudience (pane, userRoles) {
```

Comment:
Somewhat of a nitpick, but I think it'd both be clearer readability-wise and better in terms of memory allocation to not have the definition of `isMatchingAudience` enclosed in `filterAvailablePanes`.